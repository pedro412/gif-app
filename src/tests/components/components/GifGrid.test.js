import React from 'react';
import '@testing-library/jest-dom';
import { useFetchGifs } from '../../../hooks/useFetchGifs';
const { shallow } = require('enzyme');
const { default: GifGrid } = require('../../../components/GifGrid');
jest.mock('../../../hooks/useFetchGifs');

describe('Pruebas en el <GifGrid />', () => {
  const category = 'One punch';

  test('Debe mostrarse correctamente', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe mostrar items cuando se cargan imagenes con useFetchGifs', () => {
    const gifs = [
      {
        id: 'asd',
        url: 'https://localhost:/cualquiercosa.jpg',
        title: 'fake title',
      },
      {
        id: '123',
        url: 'https://localhost:/cualquiercosa.jpg',
        title: 'fake title',
      },
    ];

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });

    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
  });
});
