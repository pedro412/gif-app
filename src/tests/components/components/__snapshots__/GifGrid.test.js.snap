// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <GifGrid /> Debe mostrar items cuando se cargan imagenes con useFetchGifs 1`] = `
<div>
  <h3>
    One punch
  </h3>
  <GifGridItem
    id="asd"
    key="asd"
    title="fake title"
    url="https://localhost:/cualquiercosa.jpg"
  />
  <GifGridItem
    id="123"
    key="123"
    title="fake title"
    url="https://localhost:/cualquiercosa.jpg"
  />
</div>
`;

exports[`Pruebas en el <GifGrid /> Debe mostrarse correctamente 1`] = `
<div>
  <h3>
    One punch
  </h3>
  <p>
    Loading...
  </p>
</div>
`;
